src/sedcli_main.c

```diff
@@ -277,50 +277,31 @@ int setpw_handle_opts(char *opt, char **arg)
 
 static int get_password(struct sed_key *pwd)
 {
-	int c, flag = 0;
+	size_t dest = SED_MAX_KEY_LEN + 2;
+	uint8_t temp[dest];
 
 	echo_disable();
 
-	if (fgets((char *) pwd->key, SED_MAX_KEY_LEN, stdin) == NULL) {
+	memset(temp, 0, dest);
+
+	if (fgets((char *) temp, dest, stdin) == NULL) {
 		sedcli_printf(LOG_INFO, "Error getting password\n");
 		return -EINVAL;
 	}
-
 	sedcli_printf(LOG_INFO, "\n");
 
-	pwd->len = strnlen((char *) pwd->key, SED_MAX_KEY_LEN);
-
-	/* Handle case when user entered more characters than allowed max size */
-	if (pwd->len == SED_MAX_KEY_LEN - 1 && pwd->key[pwd->len - 1] != '\n') {
```

Comment:
This logic is needed to make sure that the input buffer has been fully read

Reply:
Here we read the input from the user and check if the user has given a pin greater than allowed length, using the following condition. Hence, this shouldn't be needed.